packages/iTwinUI-react/src/core/Table/Table.test.tsx

```diff
@@ -644,6 +650,144 @@ it('should filter table', () => {
   );
 });
 
+it('should filter false values', () => {
+  const onFilter = jest.fn();
+
+  const BooleanFilter = (
+    props: TableFilterProps<Record<string, unknown>>,
+  ): JSX.Element => {
+    const [value, setValue] = React.useState<boolean | undefined>(
+      props.column.filterValue as boolean | undefined,
+    );
+    return (
+      <BaseFilter>
+        <InputGroup displayStyle='inline'>
+          <Radio
+            name='filterOption'
+            onChange={() => setValue(true)}
+            label='True'
+            defaultChecked={value}
+          />
+          <Radio
+            name='filterOption'
+            onChange={() => setValue(false)}
+            label='False'
+            defaultChecked={value === false}
+          />
+        </InputGroup>
+        <FilterButtonBar
+          setFilter={() => props.setFilter(value)}
+          clearFilter={props.clearFilter}
+        />
+      </BaseFilter>
+    );
+  };
+
+  const columns = [
+    {
+      Header: 'Header',
+      columns: [
+        {
+          id: 'name',
+          Header: 'Name',
+          accessor: 'name',
+        },
+        {
+          id: 'booleanValue',
+          Header: 'Bool Value',
+          accessor: 'booleanValue',
+          Filter: BooleanFilter,
+          filter: 'equals',
+        },
+      ],
+    },
+  ];
+
+  const data = [
+    { name: 'Name1', description: 'Description1', booleanValue: true },
+    { name: 'Name2', description: 'Description2', booleanValue: false },
+  ] as TestDataType[];
+
+  renderComponent({ columns, onFilter, data });
+
+  userEvent.click(screen.getByRole('button'));
```

Comment:
Maybe change to querying filter icon button and clicking it because now it is not clear what `button` is.